Prox/Prox/Models/Place.swift

```diff
@@ -451,25 +472,30 @@ struct OpenHours {
     }
 
     func nextOpeningTime(forTime time: Date) -> String? {
-        guard let openingTime = getOpeningTime(forDate: time) else {
-                return nil
-        }
+        guard let allOpeningTimes = getOpeningTimes(forDate: time),
+            let openingTime = getNextOpeningPeriod(fromOpeningPeriods: allOpeningTimes, forDate: time) else {
 
-        if time < openingTime {
-            return timeString(forDate: openingTime)
+                guard let tomorrowsOpeningPeriods = getOpeningTimes(forDate: time + AppConstants.ONE_DAY),
```

Comment:
nit: maybe I'm a noob, but nesting guards is mind-bendingly confusing.

`guard` feels good to me when it's used as an early return for error states (e.g. arguments are unexpectedly nil) but requires mental overhead when used for other situations because I have to take what's written and reverse it.